### PR TITLE
Answer: 3 Directive-enchancement

### PR DESCRIPTION
**Pull request by balaji-balamurugan**
create empty, loading template, trackby function or trackby key for default trackby function.

### CODE REVIEW

**Comment by balaji-balamurugan** [comment]
Hi @tomalaforge, can you please review this pr?

File: apps/ngfor-enhancement/src/app/app.component.ts [hunk 1]
```diff
@@ -1,25 +1,128 @@
...
+        `trackBy must be a function, but received ${JSON.stringify(
+          trackByFnOrKey
+        )}.`
+      );
+    }
+    this.ngFor.ngForTrackBy =
+      typeof trackByFnOrKey !== 'function'
+        ? (i: number, a: T) => a[trackByFnOrKey]
+        : trackByFnOrKey;
+  }
```
Comment (tomalaforge) [inline apps/ngfor-enhancement/src/app/app.component.ts:71]: why not using the implementation of the trackby function inside ngFor ?

**Comment by tomalaforge** [comment]
very nice implementation. work well. nice work👍

File: apps/ngfor-enhancement/src/app/app.component.ts (hunk 1 above)
Comment (balaji-balamurugan) [reply apps/ngfor-enhancement/src/app/app.component.ts:71]: ngFor doesn't have a trackByKey option, so I tried implement it here.

Comment (tomalaforge) [reply apps/ngfor-enhancement/src/app/app.component.ts:71]: ah yes I didn't see the trackByKey 👍

**Pull request closed without merging**